### ISSUE TITLE
How to freeze the weight of mask?

### ISSUE THREAD

**Issue by liz2020** [facebookresearch/maskrcnn-benchmark#1001]
## ❓ Questions and Help
I have trained the model on one dataset, but I want to keep the weight of the mask unchanged when training on another dataset. What parameter do I need to change, or what part of code do I need to modify? Thanks

**Comment by Dorozhko-Anton** [comment]
If you only want to freeze `roi_heads.mask` weights you can modify `make_optimizer` used here https://github.com/facebookresearch/maskrcnn-benchmark/blob/24c8c90efdb7cc51381af5ce0205b23567c3cd21/tools/train_net.py#L41

and skip parameters with `key` that contains `mask`  or `roi_heads.mask`
https://github.com/facebookresearch/maskrcnn-benchmark/blob/24c8c90efdb7cc51381af5ce0205b23567c3cd21/maskrcnn_benchmark/solver/build.py#L7

**Comment by liz2020** [comment]
Thank you very much, I will try this out.

**Comment by hoonkai** [comment]
> If you only want to freeze `roi_heads.mask` weights you can modify `make_optimizer` used here
> 
> https://github.com/facebookresearch/maskrcnn-benchmark/blob/24c8c90efdb7cc51381af5ce0205b23567c3cd21/tools/train_net.py#L41
> 
> and skip parameters with `key` that contains `mask` or `roi_heads.mask`
> 

Doesn't `requires_grad=False` need to be set for all parameters to be frozen? Or does excluding the param from the optimiser suffice?

**Comment by liz2020** [comment]
Only the params included in the optimizer will be updated. So excluding the param suffice. [See here for detail](https://discuss.pytorch.org/t/how-the-pytorch-freeze-network-in-some-layers-only-the-rest-of-the-training/7088/22)

**Comment by hoonkai** [comment]
> Only the params included in the optimizer will be updated. So excluding the param suffice. [See here for detail](https://discuss.pytorch.org/t/how-the-pytorch-freeze-network-in-some-layers-only-the-rest-of-the-training/7088/22)

Thanks
